cluster/network_config.go

```diff
@@ -25,31 +25,74 @@ import (
 	"github.com/hazelcast/hazelcast-go-client/types"
 )
 
-const defaultAddress = "127.0.0.1:5701"
+const (
+	defaultAddress      = "127.0.0.1:5701"
+	defaultPortRangeMin = 5701
+	defaultPortRangeMax = 5703
+)
 
 type NetworkConfig struct {
 	SSL               SSLConfig      `json:",omitempty"`
 	Addresses         []string       `json:",omitempty"`
+	PortRange         PortRange      `json:",omitempty"`
 	ConnectionTimeout types.Duration `json:",omitempty"`
 }
 
+type PortRange struct {
+	Min int `json:",omitempty"`
+	Max int `json:",omitempty"`
+}
+
+func (pr *PortRange) Clone() PortRange {
+	return PortRange{
+		Min: pr.Min,
+		Max: pr.Max,
+	}
+}
+
 func (c *NetworkConfig) Clone() NetworkConfig {
 	addrs := make([]string, len(c.Addresses))
 	copy(addrs, c.Addresses)
 	return NetworkConfig{
 		Addresses:         addrs,
 		ConnectionTimeout: c.ConnectionTimeout,
 		SSL:               c.SSL.Clone(),
+		PortRange:         c.PortRange.Clone(),
+	}
+}
+
+func (c *NetworkConfig) SetPortRange(min int, max int) {
+	c.PortRange = PortRange{
+		Min: min,
+		Max: max,
 	}
 }
 
 func (c *NetworkConfig) Validate() error {
+	// set port range defaults if needed
+	if c.PortRange.Min == 0 && c.PortRange.Max == 0 {
+		c.PortRange = PortRange{
+			Min: defaultPortRangeMin,
+			Max: defaultPortRangeMax,
+		}
+	}
+
+	// validate port range
+	if err := c.validatePortRange(); err != nil {
+		return err
+	}
+
+	// validate addresses
```

Comment:
Is this comment necessary?

Reply:
in my opinion yes, to mark the portion of the code will be do the address validation part